smart_contracts/contract/src/contract_api/storage.rs

```diff
@@ -134,7 +136,31 @@ pub fn new_contract(
     hash_name: Option<String>,
     uref_name: Option<String>,
 ) -> (ContractHash, ContractVersion) {
-    let (contract_package_hash, access_uref) = create_contract_package_at_hash();
+    create_contract(entry_points, named_keys, hash_name, uref_name, false)
+}
+
+/// Create a locked contract stored under a Key::Hash, which can never be upgraded. This is an
+/// irreversible decision; for a contract that can be upgraded use `new_contract` instead.
+/// if `named_keys` are provided, will apply them
+/// if `hash_name` is provided, puts contract hash in current context's named keys under `hash_name`
+/// if `uref_name` is provided, puts access_uref in current context's named keys under `uref_name`
+pub fn new_locked_contract(
```

Comment:
As said earlier that's ok to add an argument of `ContractPackageStatus` to `new_contract` (and in `create_contract_package_at_hash`) here so we don't have two functions: `new_contract` and `new_locked_contract`

Reply:
I asked him to do it this way to avoid breaking pre-existing contracts, and also to keep the semantics clear and call out the irrevocable choice being made in the documentation.

Reply:
Understood